### PR TITLE
Exercise rights and intial state

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

**Comment by krulis-martin** [comment]
Ignore Travis Errors for now, I still need to fix tests (as some assumptions were changed).

File: app/V1Module/security/Policies/ReferenceExerciseSolutionPermissionPolicy.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+use App\Model\Entity\ReferenceExerciseSolution;
+use App\Security\Identity;
+
+class ReferenceExerciseSolutionPermissionPolicy implements IPermissionPolicy {
+  public function getAssociatedClass() {
+    return ReferenceExerciseSolution::class;
+  }
+
+  public function isAuthor(Identity $identity, ReferenceExerciseSolution $referenceExerciseSolution = null) {
+    if ($referenceExerciseSolution === null) {
```
Comment (Teyras) [inline app/V1Module/security/Policies/ReferenceExerciseSolutionPermissionPolicy.php]: We use `NULL`, not `null`. The same goes for `true` and `false`.

Comment (krulis-martin) [reply app/V1Module/security/Policies/ReferenceExerciseSolutionPermissionPolicy.php]: That is in violation of http://www.php-fig.org/psr/psr-2/

Comment (krulis-martin) [reply app/V1Module/security/Policies/ReferenceExerciseSolutionPermissionPolicy.php]: null, true and false MUST be in lowercase

Comment (Teyras) [reply app/V1Module/security/Policies/ReferenceExerciseSolutionPermissionPolicy.php]: Well, if you want to search for violations of PSR, there are plenty more :) IMO it's more important to be consistent... But this is a minor nitpick and it's possible we don't follow that rule so strictly anyway.

Comment (krulis-martin) [reply app/V1Module/security/Policies/ReferenceExerciseSolutionPermissionPolicy.php]: Consistency is broken anyways. Lets shift this issue into another task (which will ensure consistency in one way or another).

Comment (Teyras) [reply app/V1Module/security/Policies/ReferenceExerciseSolutionPermissionPolicy.php]: Agreed, have my approval :)

**Pull request merged**